src/main/java/cz/brmlab/yodaqa/pipeline/structured/StructuredPrimarySearch.java

```diff
@@ -179,6 +197,19 @@ protected void propertyToAnswer(JCas jcas, PropertyValue property,
 
 		ai.addToIndexes();
 	}
+	protected int getSourceID(String url, String label, JCas questionView){
+		if (sourceIDs.containsKey(url)) {
+			return sourceIDs.get(url);
+		}
+		else {
```

Comment:
When most of your code is in an else {} branch, maybe you should've just dropped the else.